InteractiveHtmlBom/ecad/genericjson.py

```diff
@@ -0,0 +1,73 @@
+import io
+import json
+from jsonschema import validate, ValidationError
+
+from .common import EcadParser, Component
+
+
+class GenericJsonParser(EcadParser):
+    COMPATIBLE_SPEC_VERSIONS = [1]
+
+    def get_generic_json_pcb(self):
+        from os import path
+        with io.open(self.file_name, 'r') as f:
+            pcb = json.load(f)
+
+        if 'spec_version' not in pcb:
+            raise ValidationError("'spec_version' is a required property")
+
+        if pcb['spec_version'] not in self.COMPATIBLE_SPEC_VERSIONS:
+            raise ValidationError("Unsupported spec_version ({})"
+                                  .format(pcb['spec_version']))
+
+        schema_dir = path.join(path.dirname(__file__), 'schema')
+        schema_file_name = path.join(schema_dir,
+                                     'genericjsonpcbdata_v{}.schema'
+                                     .format(pcb['spec_version']))
+
+        with io.open(schema_file_name, 'r') as f:
+            schema = json.load(f)
+
+        validate(instance=pcb, schema=schema)
+
+        return pcb
+
+    def _verify(self, pcb):
+
+        """Spot check the pcb object."""
+
+        if len(pcb['pcbdata']['footprints']) != len(pcb['components']):
+            self.logger.error("length of components list doesn't match"
```

Comment:
nit: error messages sould be full sentences. I.e. capitalize first letter, end with a period.